supabase/functions/_shared/helpers.ts

```diff
@@ -1,20 +1,51 @@
-import { corsHeaders } from "./cors.ts";
+import HandlebarsJS from 'https://esm.sh/handlebars';
+import Mustache from 'https://esm.sh/mustache';
+import { corsHeaders } from './cors.ts';
 
 export const returnPostgresError = (error: any) => {
-  return new Response(JSON.stringify({ error }), {
-    headers: {
-      ...corsHeaders,
-      "Content-Type": "application/json",
-    },
-    status: 400,
-  });
+    return new Response(JSON.stringify({ error }), {
+        headers: {
+            ...corsHeaders,
+            'Content-Type': 'application/json',
+        },
+        status: 400,
+    });
 };
 
 export const handlebarsHelpers = {
-  urlencode: function(s: string) {
-    return encodeURIComponent(s)
-  },
-  basicauth: function(user: string, password: string) {
-    return btoa(`${user}:${password}`);
-  }
-}
+    urlencode: function (s: string) {
+        return encodeURIComponent(s);
+    },
+    basicauth: function (user: string, password: string) {
+        return btoa(`${user}:${password}`);
+    },
+};
+
+export const compileTemplate = (template: string, data: any, connector_id: string) => {
+    (HandlebarsJS as any).registerHelper(handlebarsHelpers);
+
+    const compiledTemplate = (HandlebarsJS as any).compile(template);
+    const handlebarsOutput = compiledTemplate(data);
+
+    try {
+        // Override escaping globally for mustache
+        (Mustache.escape as any) = function (text: any) {
+            return text;
+        };
+
+        //Duplicate for testing Mustache
+        const mustacheOutput = Mustache.render(template, {
+            ...data,
+            ...handlebarsHelpers,
+        });
+
+        if (handlebarsOutput !== mustacheOutput) {
+            console.log(`oauth:templates:mismatch:${connector_id}`, { template });
+        }
+    } catch (e) {
+        // This try catch is just to be super safe
+        console.log(`oauth:templates:error:${connector_id}`, e);
+    }
```

Comment:
The logging of the `template` should end up truncated. I am thinking a truncated template and the `connector_id` should be enough to find it in the DB. However, we could log the full one if folks think that is needed.

Decided to be super safe and just wrap in a try catch. Not really sure what could cause this but do not want to blow up prod on accident with these changes.